src/utils/ethereumUtils.js

```diff
@@ -515,7 +524,35 @@ async function parseEthereumUrl(data) {
   });
 }
 
+const calculateL1FeeOptimism = async (tx, provider) => {
+  try {
+    tx.value = toHex(tx.value);
+    tx.nonce = Number(await provider.getTransactionCount(tx.from));
```

Comment:
hmmm

Reply:
will this be affected by chris' nonce mgmt stuff?

Reply:
this is not a problem in optimism because txs are confirmed way faster than you can complete any flow.
However, after #2532 is merged I can get the nonce through the nonce manager instead